Source/WTF/wtf/NativePromise.h

```diff
@@ -420,6 +420,16 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
         dispatchAll(lock);
     }
 
+    template<typename SettleValueType>
+    void settleChainedPromise(SettleValueType&& result, const Logger::LogSiteIdentifier& site)
```

Comment:
Using twice settleChainedPromise name for two different methods makes things more difficult to read than it should.
How about renaming this one to `settleAsChainedPromise`.